Source/CBLMultipartReader.m

```diff
@@ -137,17 +138,20 @@ - (BOOL) parseHeaders: (NSString*)headersStr {
 
 
 - (NSRange) searchFor: (NSData*)pattern from: (NSUInteger)start {
-    return [_buffer rangeOfData: pattern
+    if (!_buffer.hasBytesAvailable)
+        return NSMakeRange(NSNotFound, 0);
+    NSData *data = [NSData dataWithBytesNoCopy:_buffer.mutableBytes length:_buffer.bytesAvailable freeWhenDone:NO];
```

Comment:
Yea it could be implemented directly there. To be honest I copied my cyclic buffer from `PocketSocket` and just modified it a bit to suite your comments. I don't think there will be a huge performance difference here as it doesn't copy the bytes.

I was trying to not wholesale copy all the NSData API's over but rather just make a more workable `void *`.